python/lsst/dax/apdb/apdbSql.py

```diff
@@ -387,6 +387,35 @@ def getDiaForcedSources(
         _LOG.debug("found %s DiaForcedSources", len(sources))
         return sources
 
+    def contains(self, visit: int, detector: int) -> bool:
+        # docstring is inherited from a base class
+        raise NotImplementedError()
+
+    def containsCcdVisit(self, ccdVisitId: int) -> bool:
+        """Test whether data for a given visit-detector is present in the APDB.
+
+        This method is a placeholder until `Apdb.contains` can be implemented.
+
+        Parameters
+        ----------
+        ccdVisitId : `int`
+            The packed ID of the visit-detector to search for.
+
+        Returns
+        -------
+        present : `bool`
+            `True` if some DiaSource records exist for the specified
+            observation, `False` otherwise.
+        """
+        # TODO: remove this method in favor of contains on DM-?????
+        table: sqlalchemy.schema.Table = self._schema.get_table(ApdbTables.DiaSource)
+        # Query should only load one page of the index
```

Comment:
Well, if it guesses right 🙂 Usually finding data in index needs index scan which can bring multiple pages into memory. But it will not need to read any tuple pages, that is true. This should be efficient enough.

Reply:
Will reword to "one leaf page" (the current implementation, at least, uses B-trees for all indices).